iroh-unixfs/src/builder.rs

```diff
@@ -559,6 +598,10 @@ impl DirectoryBuilder {
 
         Ok(if let Some(path) = path {
             let mut dir = make_dir_from_path(path, chunker.clone(), degree).await?;
+            match &mut dir {
+                Directory::Basic(basic) => basic.entries.extend(entries),
+                Directory::Hamt(_) => unimplemented!(),
```

Comment:
`Hamt`s are implemented, so this shouldn't be the case

Reply:
Ok, I need to figure out how to add entries to existent HamtNode. Probably worth to rebuild it entirely

Reply:
The way this builder works is a bit weird. If path is set, entries are ignored. So maybe you can just put in an assertion that either path is None or entries is empty and be done with it?

I tried to refactor all of this at some point, but was not entirely happy with the results.

Reply:
Afaik there are no stoppers for working both, `path` and `entries`. It is just implemented this way for some reasons. I may try to refactor this part

Reply:
@rklaehn Done with refactoring. I removed path variable in builder and added add_path method that walks over path, converts elements to `Entry` and adds them to `self.entries`. It makes API more consistent and allows to add multiple paths.